### ISSUE TITLE
unknown url type

### ISSUE THREAD

**Issue by marshaul** [mps-youtube/mps-youtube#593]
Overall mpsyt has been working for me, except the following single video:

https://www.youtube.com/watch?v=ycnDw1kiG50

This gives the error message "unknown url type: '/yts/jsbin/player-en_US-vflg9Wu9U/base.js'"

**Comment by ids1024** [comment]
Does `mpsyt --version` show the `youtube-dl` backend? It should.

(I should probably make it require explicit configuration to use the internal backend, even if youtube-dl is not installed).

**Comment by jkleve** [comment]
I just started getting this yesterday too. `mpsyt --version` gave the warning `youtube-dl` not found. I have fedora so I did `dnf install youtube-dl` and after that mpsyt was back to normal.

**Comment by marshaul** [comment]
The problem is definitely not having the youtube-dl backend properly in place. In my case, I planned to use Homebrew (I'm on a Mac) to install everything, so I installed youtube-dl that way, but when it got to mpsyt it choked because my Homebrew only has python2 linked into my PATH. 

Rather than change my Homebrew setup around, I just went ahead and installed mpsyt directly via pip3. Of course, doing this mpsyt fails to locate youtube-dl. So I went ahead and reinstalled youtube-dl via pip3 and it works properly now.

Thanks!

**Comment by ids1024** [comment]
Yes; the internal back-end is fairly broken; I've decided to disable it and require manual override by setting an environmental variable to use it.

https://github.com/mps-youtube/pafy/commit/726c1a77c024bf1048dfe824d79e43aed227ecfe changes this. It will be in the next Pafy release.

I was originally going to just get rid of the internal back-end in favor of youtube-dl (pafy originally had only the internal one) but a few people complained.

**Comment by 2E0PGS** [comment]
Some reason my mpsyt wont detect my youtube-dl could that be because I installed it using python-pip not python-pip3 ?
